jsr352/jberet/src/main/java/org/hibernate/search/jsr352/jberet/impl/HibernateSearchJsr352Extension.java

```diff
@@ -0,0 +1,124 @@
+import org.hibernate.search.jsr352.massindexing.impl.steps.lucene.EntityReader;
+import org.jberet.cdi.JobScoped;
+import org.jberet.cdi.PartitionScoped;
+
+
+/**
+ * A CDI extension to explicitly add hibernate-search-jsr352-core types to the CDI context.
+ * <p>
+ * These types must be registered into the CDI context because they expect one of their
+ * attributes to be injected, thus they need to be instantiated by CDI.
+ * <p>
+ * Ideally we should use component scanning to that effect, but:
+ * <ul>
+ * <li>In CDI, component scanning is configured per-module in META-INF/beans.xml,
+ * with no way to add a package from another module.
+ * <li>In Wildfly, component scanning in dependencies is always done with the "all" discovery type,
+ * meaning that even components without CDI annotations will be discovered.
+ * Besides exposing private types unnecessarily,
+ * this is annoying because it puts default {@link EntityManagerFactoryRegistry} implementations into the CDI context,
+ * leading to conflicts when injecting such types
+ * (since this module, hibernate-search-jsr352-jberet, also provides an implementation).
+ * <br>See <a href="https://issues.jboss.org/browse/WFLY-8656">WFLY-8656</a>.
+ * <li>Even if we could solve the above, JBeret requires specific scope annotations to be set on the
+ * beans so that they are correctly scoped. Since the core isn't dependent on JBeret, we must specify these
+ * scopes manually when registering the beans.
+ * </ul>
+ * Thus we use explicit type and scope registration as a workaround.
+ *
+ * @author Yoann Rodiere
+ */
+public class HibernateSearchJsr352Extension implements Extension {
+
+	public void afterTypeDiscovery(@Observes AfterTypeDiscovery event, BeanManager beanManager) {
+		registerType( event, beanManager, JobContextSetupListener.class );
+		registerType( event, beanManager, EntityReader.class );
+	}
+
+	public void afterBeanDiscovery(@Observes AfterBeanDiscovery event, BeanManager beanManager) {
+		addScopeAlias( event, beanManager, HibernateSearchJobScoped.class, JobScoped.class );
+		addScopeAlias( event, beanManager, HibernateSearchPartitionScoped.class, PartitionScoped.class );
+	}
+
+	private void registerType(AfterTypeDiscovery event, BeanManager beanManager, Class<?> clazz) {
+		AnnotatedType<?> annotatedType = beanManager.createAnnotatedType( clazz );
+		event.addAnnotatedType( annotatedType, clazz.getName() );
+	}
+
+	private void addScopeAlias(AfterBeanDiscovery event, BeanManager beanManager,
+			Class<? extends Annotation> alias, Class<? extends Annotation> target) {
+		event.addContext( new AliasedContext( alias, beanManager, target ) );
+	}
+
+	private static class AliasedContext implements Context {
+
+		private final Class<? extends Annotation> scopeType;
+		private final BeanManager targetBeanManager;
+		private final Class<? extends Annotation> targetScopeType;
+
+		public AliasedContext(Class<? extends Annotation> scopeType,
```

Comment:
It won't change anything: the class itself being private, the constructor won't be available outside of the surrounding class anyway. I feel making the constructor public makes it clearer that the surrounding class is expected to use it, though I'll grant you it won't change anything in practice (since the surrounding class is allowed to access private elements in the nested class, which is absurd if you ask me).

To sum up, it's a matter of taste...